File: Easy/1.twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map =new Map();
    for(i=0;i<nums.length;i++){
        let cmpl = target-nums[i];
        if(map.has(cmpl))
            return[map.get(cmpl),i];
        map.set(nums[i],i)
    }
    return[]
};

File: Easy/1.twosum.py
1. # 1. Brute force approach 
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i]+nums[j]==target :
                    return [i,j]
                
                
#2. Hashmap approach
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for i in range(len(nums)):
           find = target-nums[i]
           if find in seen:

            return [seen[find], i]

           seen[nums[i]] = i
